src/vm/i386/asmhelpers.S

```diff
@@ -392,8 +399,14 @@ LEAF_END ArrayOpStubTypeMismatchException, _TEXT
 // ------------------------------------------------------------------------------
 //  void STDCALL CallDescrWorkerInternal(CallDescrWorkerParams *  pParams)
 NESTED_ENTRY CallDescrWorkerInternal, _TEXT, NoHandler
+    PROLOG_BEG
+    PROLOG_PUSH eax
```

Comment:
You don't need to push EAX, EDX and ECX here. They are not expected to be preserved. I have taken a look at a C function compiled by both clang and gcc and they don't preserve these registers either (and use them)